### PR TITLE
Pass X-Presto-Client-Info in presto hook

### PR DESCRIPTION
**Pull request by pingzh**
so that in the presto side, it can track back to an airflow task, according to this doc https://prestodb.io/docs/current/develop/client-protocol.html

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/presto/hooks/presto.py [hunk 1]
```diff
@@ -27,6 +28,24 @@
...
+    context_var = {
+        format_map['default'].replace(DEFAULT_FORMAT_PREFIX, ''): os.environ.get(
+            format_map['env_var_format'], ''
+        )
+        for format_map in AIRFLOW_VAR_NAME_FORMAT_MAPPING.values()
+    }
+    task_info = {
+        'dag_id': context_var['dag_id'],
+        'task_id': context_var['task_id'],
+        'execution_date': context_var['execution_date'],
```
Comment (potiuk) [inline airflow/providers/presto/hooks/presto.py:53]: You should have ruin _id here as well.

Comment (pingzh) [reply airflow/providers/presto/hooks/presto.py:53]: good idea. added

**Comment by pingzh** [comment]
@potiuk any suggestions that I can fix this https://github.com/apache/airflow/runs/5648830794?check_suite_focus=true#step:11:1039

since there is not `DEFAULT_FORMAT_PREFIX` in 2.1.1 https://github.com/apache/airflow/blob/2.1.1/airflow/utils/operator_helpers.py#L20

```
There were errors when importing Providers!
This test is run to check Provider's compatibility with Airflow 2.1.0
So this error might mean that your providers are not Airflow 2.1.0 compatible

Detailed errors:

----------------------------------------
Exception when importing: airflow.providers.google.cloud.example_dags.example_presto_to_gcs


Traceback (most recent call last):
  File "/opt/airflow/dev/import_all_classes.py", line 90, in import_all_classes
    _module = importlib.import_module(modinfo.name)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name, package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/google/cloud/example_dags/example_presto_to_gcs.py", line 32, in <module>
    from airflow.providers.google.cloud.transfers.presto_to_gcs import PrestoToGCSOperator
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/google/cloud/transfers/presto_to_gcs.py", line 24, in <module>
    from airflow.providers.presto.hooks.presto import PrestoHook
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/presto/hooks/presto.py", line 31, in <module>
    from airflow.utils.operator_helpers import AIRFLOW_VAR_NAME_FORMAT_MAPPING, DEFAULT_FORMAT_PREFIX
ImportError: cannot import name 'DEFAULT_FORMAT_PREFIX' from 'airflow.utils.operator_helpers' (/usr/local/lib/python3.7/site-packages/airflow/utils/operator_helpers.py)
```

**Comment by potiuk** [comment]
Jus don't import it from there. or catch an import and provide a suitable default. I do not know the history of DEFAULT_FORMAT_PREFIX so cannot give a definite answer, but definitely if you add this import - you are relying on 2.2 behaviour, which we should not do. 
According to https://github.com/apache/airflow/blob/main/README.md#support-for-providers all the providers until 21st of May should ne 2.1-compatible.

**Comment by potiuk** [comment]
Seems that it's you who added the prefix, so you should know what to use in the provider to make it 2.1 comatible I think :)

**Comment by potiuk** [comment]
Seems that the prefix was actually added in 2.3 so you MUST make it compatible because 2.2 compatibility will be there for quite a while. The 2.2 compatibility is there till October 11.

File: airflow/providers/presto/hooks/presto.py [hunk 2]
```diff
@@ -27,6 +28,33 @@
...
+def generate_presto_client_info() -> str:
+    """Return json string with dag_id, task_id, execution_date and try_number"""
+    context_var = {
+        format_map['default'].replace(DEFAULT_FORMAT_PREFIX, ''): os.environ.get(
+            format_map['env_var_format'], ''
+        )
+        for format_map in AIRFLOW_VAR_NAME_FORMAT_MAPPING.values()
+    }
+    task_info = {
+        'dag_id': context_var['dag_id'],
```
Comment (eladkal) [inline airflow/providers/presto/hooks/presto.py:51]: Can we add also the dag owner?

Comment (pingzh) [reply airflow/providers/presto/hooks/presto.py:51]: 👍 added

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by eladkal** [comment]
@pingzh Should we do that also for TrinoHook?
https://trino.io/docs/current/develop/client-protocol.html#client-request-headers

**Comment by pingzh** [comment]
> @pingzh Should we do that also for TrinoHook? https://trino.io/docs/current/develop/client-protocol.html#client-request-headers

@eladkal good catch. let me add that as well.

**Pull request merged**